workers/unity/Assets/Playground/Scripts/Cubes/CollisionProcessSystem.cs

```diff
@@ -0,0 +1,76 @@
+using Generated.Playground;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Playground
+{
+    [RemoveAtEndOfTick]
+    public struct CollisionComponent : IComponentData
+    {
+        public Entity ownEntity;
+        public Entity otherEntity;
+    }
+
+    [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
+    internal class CollisionProcessSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            public readonly int Length;
+            public EntityArray Entities;
+            public ComponentDataArray<SpatialOSLaunchable> Launchable;
```

Comment:
Having both this and the command sender is generally an anti-pattern. You want each system to have exactly the data it needs, both to prevent performance loss and, more importantly, to make it easy to tell at a glance what the system does. You don't want to include more stuff than you need and work out which bits to write to later if you can avoid it.

Reply:
The way to avoid this here is for the collision detection code to work out what is going on and add the relevant component to each entity. Then the presents of a specific component tells you whether you want to write to state or send a command. This could also be done in a system that runs before this if you wanted to keep the responsibility of each system / monobehaviour as limited as possible.